src/picongpu/include/fields/background/templates/bgrTWTS.def

```diff
@@ -0,0 +1,263 @@
+            
+            /** Obtain the SI time that later enters the Ex(r, t) calculations as r.
+             * \tparam T_dim Specializes for the simulation dimension
+             *  \param CurrentStep current time step number of simulation
+             *  \return time in SI units */
+            template <unsigned T_dim>
+            HDINLINE float_64 getTime_SI(const uint32_t currentStep) const;
+            
+            HDINLINE PMacc::math::Vector<float3_X,numComponents>
+            dummyCast(const PMacc::math::Vector<float3_X,numComponents>& x) const;
+            HDINLINE PMacc::math::Vector<float3_X,numComponents>
+            dummyCast(const PMacc::math::Vector<float2_X,numComponents>& x) const;
+            HDINLINE float3_X
+            dummyCast(const float3_X& x) const { return x; }
+            HDINLINE float3_X
+            dummyCast(const float2_X& x) const { return float3_X( x.x(),x.y(),float_X(0.0) ); }
+        
+        };
+
+        class TWTSFieldB
+        {
+        public:
+            typedef float_X float_T;
+            
+            /* Center of simulation volume in number of cells */
+            PMACC_ALIGN(halfSimSize,DataSpace<simDim>);
+            /* y-position of TWTS coordinate origin inside the simulation coordinates [meter]
+               The other origin coordinates (x and z) default to globally centered values
+               with respect to the simulation volume. */
+            const PMACC_ALIGN(focus_y_SI,float_64);
+            /* Laser wavelength [meter] */
+            const PMACC_ALIGN(wavelength_SI,float_64);
+            /* TWTS laser pulse duration [second] */
+            const PMACC_ALIGN(pulselength_SI,float_64);
+            /* line focus height of TWTS pulse [meter] */
+            const PMACC_ALIGN(w_x_SI,float_64);
+            /* line focus width of TWTS pulse [meter] */
+            const PMACC_ALIGN(w_y_SI,float_64);
+            /* interaction angle between TWTS laser propagation vector and the y-axis [rad] */
+            const PMACC_ALIGN(phi,float_X);
+            /* propagation speed of TWTS laser overlap
+               normalized to the speed of light. [Default: beta0=1.0] */
+            const PMACC_ALIGN(beta_0,float_X);
+            /* If auto_tdelay=FALSE, then a user defined delay is used. [second] */
+            const PMACC_ALIGN(tdelay_user_SI,float_64);
+            /* Make time step constant accessible to device. */
+            const PMACC_ALIGN(dt,float_64);
+            /* Make length normalization constant accessible to device. */
+            const PMACC_ALIGN(unit_length,float_64);
+            /* Should the TWTS laser delay be chosen automatically, such that
+            the laser gradually enters the simulation volume? [Default: TRUE] */
+            const PMACC_ALIGN(auto_tdelay,bool);
+            /* Number of field components used in the simulation. [Default: 3 for both 2D and 3D] */
+            const static int numComponents=DIM3;
```

Comment:
I am not sure about that: the `.hpp` files include stuff like `#include "simulation_defines.hpp"` but this very file right now is a `def` file which is used in `param` files (besides that a `Fields.hpp` does not exist due to that reason).